packages/xo-server/src/api/vm.js

```diff
@@ -1137,10 +1137,15 @@ resume.resolve = {
 // -------------------------------------------------------------------
 
 export async function revert({ snapshot, snapshotBefore }) {
-  await this.checkPermissions(this.user.id, [
-    [snapshot.$snapshot_of, 'operate'],
-  ])
-  return this.getXapi(snapshot).revertVm(snapshot._xapiId, snapshotBefore)
+  const { id: userId, permission } = this.user
+  await this.checkPermissions(userId, [[snapshot.$snapshot_of, 'operate']])
+  const vmSnapshot = await this.getXapi(snapshot).revertVm(
```

Comment:
`newSnapshot`?
Reading code with `snapshot` and `vmSnapshot` is confusing: they both "mean" the same thing and we don't know which is which.